Implement incidenttypes components

mkumbobeaty opened the pull request.

mkumbobeaty committed:
feat(incidents): create incidenttypes components

create the components for incidentstypes and replacate the contact list to the incidenttypes list

mkumbobeaty committed:
refactor(incidenttypes): refactor contents in incidenttypes

change and remove unwanted contents in incidenttypes components

mkumbobeaty committed:
feat(incidenttypes): intergrate with api

intergate incidenttypes component with api,get the data and display them

mkumbobeaty committed:
fix(incidenttypes): fix incidenttypes pagination

add get method from api which help in paginating incidenttypes

mkumbobeaty committed:
Merge branch 'develop' into feature/incident_types

mkumbobeaty committed:
feat(incidenttypes): add search functionality

improve search by adding functionality for searching incidenttypes

BenMaruchu commented on src/Incidents/components/IncidentTypes/ActionBar/index.js:
```diff
@@ -0,0 +1,92 @@
+        <Button
+          type="circle"
+          icon="cloud-download"
+          title="Export selected incidenttypes"
```
This is a title supposed to be `Export selected Incident Types`

BenMaruchu replied:
So for other places for title attribute change `incidenttypes`  to be a normal word

mkumbobeaty replied:
oky will fix that

BenMaruchu reviewed (changes requested):
@mkumbobeaty hey there a some few quick update for this PR thanks

mkumbobeaty committed:
refactor(incidenttypes): improve title

improve title by changing incidenttypes to be a normal word(i.e incident types)

BenMaruchu reviewed (changes requested):
@mkumbobeaty  sorry for disturbing but can you make first letters for incident types caps i.e `Incident Types` for titles

BenMaruchu commented:
Also there is a conflict on this branch please pull changes first before pushing @mkumbobeaty

mkumbobeaty committed:
refactor(incidents): change actions title to caps

mkumbobeaty committed:
perf(incidents): merge with develop and fix conflict
